.github/workflows/tests.yaml

```diff
@@ -0,0 +1,47 @@
+name: Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+
+    name: Format check
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version-file: 'go.mod'
+
+    - name: Ensure no formatting changes
+      run: |
+        go fmt ./...
+        git diff --exit-code
+
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        go: ['1.21']
+
+    name: Go ${{ matrix.go }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Test and build
+        run: |
+          go test -race ./... -check.vv
```

Comment:
No need for the `-check.vv` argument here. That's a `gopkg.in/check.v1` feature -- Go tests using plain stdlib `testing` (which is probably what we'll add here?) don't have this flag.

Thinking out loud: yeah, on the one hand it's a bit premature to add CI to run tests when we don't have any tests yet :-), on the other hand, it certainly doesn't hurt, and will mean that CI will kick in as soon as we write some.

The code isn't set up very well for testing right now. I'd suggest you write one or two very basic tests for a couple of helper functions in your new code (not ones that deal with the network for now), just to get the feel for `go test` and to exercise this a little.